### PR TITLE
Create rounded button feature

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #212 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

Create new `rounded` button feature and add examples to docsite.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/demo.md [hunk 1]
```diff
@@ -381,6 +381,62 @@ Use the `slim` attribute along with `iconOnly` attribute on the `auro-button` el
 
 </auro-accordion>
 
+## Rounded
+
```
Comment (jason-capsule42) [inline demo/demo.md:385]: Let's add some descriptive language. We should explain how this button is different and explain the slots here since we divided them up to achieve the animation.

File: demo/demo.md [hunk 2]
```diff
@@ -381,6 +381,62 @@ Use the `slim` attribute along with `iconOnly` attribute on the `auro-button` el
...
+    Click me to hide/show text
+    <auro-icon customSize customcolor category="in-flight" name="wifi" slot="icon"></auro-icon>
+  </auro-button>
+</div>
+
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+  ```html
+  <auro-button aria-label="in-flight" id="targetButton" rounded>
```
Comment (jason-capsule42) [inline demo/demo.md:432]: It would be good to show this button left aligned (as it currently is) and right aligned. When right aligned, the animation should appear to grow to the left.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
In addition to the inline comments I am also getting a linting error when I run the build. The error is because we are linting `./src/*.css` when I think we should only be doing `./src/*.scss`.

File: demo/index.html [hunk 3]
```diff
@@ -25,6 +25,13 @@
           const rawHtml = marked.parse(text);
           document.querySelector('main').innerHTML = rawHtml;
           Prism.highlightAll();
+        })
+        .then(() => {
+          const targetButton = document.getElementById('targetButton');
```
Comment (jason-capsule42) [inline demo/index.html:30]: We need to use better selectors here. `targetButton` is very generic and could be inadvertently used elsewhere on the page. 

Also, this approach needs to be updated to use the new JS example method so it will work on the docsite.

File: docs/api.md [hunk 4]
```diff
@@ -15,10 +15,12 @@
 | `autofocus`      | `autofocus`      | `Boolean` | false   | This Boolean attribute lets you specify that the button should have input focus when the page loads, unless overridden by the user |
 | `disabled`       | `disabled`       | `Boolean` | false   | If set to true button will become disabled and not allow for interactions |
 | `fluid`          | `fluid`          | `Boolean` | false   | Alters the shape of the button to be full width of its parent container |
+| `hideText`       | `hideText`       | `Boolean` | false   | If set to true, hides default slot content       |
```
Comment (jason-capsule42) [inline docs/api.md:18]: This should say something about only for use with the `rounded` attribute.

File: docs/api.md [hunk 5]
```diff
@@ -29,9 +31,10 @@
 
 ## Events
 
-| Event              | Type               | Description                                      |
-|--------------------|--------------------|--------------------------------------------------|
-| `auroButton-ready` | `CustomEvent<any>` | Notifies that the component has finished initializing. |
+| Event                   | Type               | Description                                      |
+|-------------------------|--------------------|--------------------------------------------------|
+| `auroButton-ready`      | `CustomEvent<any>` | Notifies that the component has finished initializing. |
+| `auroButton-toggleText` |                    | Hides/shows default slot content.                |
```
Comment (jason-capsule42) [inline docs/api.md:37]:
```suggestion
| `auroButton-toggledText` |                    | Hides/shows default slot content.                |
```

File: src/auro-button.js [hunk 6]
```diff
@@ -28,22 +30,26 @@ import '@alaskaairux/auro-loader';
...
+
+/* eslint-disable max-statements, no-negated-condition */
 export class AuroButton extends LitElement {
 
-  /* eslint max-statements: ["error", 11] */
   constructor() {
     super();
     this.autofocus = false;
     this.disabled = false;
+    this.hideText = false;
```
Comment (jason-capsule42) [inline src/auro-button.js:45]: Should this be private?

File: src/style.scss [hunk 7]
```diff
@@ -408,8 +410,37 @@ slot {
...
+      height: var(--auro-size-300, 24px);
+    }
+  }
+
+  // auro-button--hideText
+  &--hideText {
+    min-width: unset;
+    padding: unset;
+
+    width: 48px;
```
Comment (jason-capsule42) [inline src/style.scss:432]: Do we have a token for this size?

File: src/style.scss [hunk 8]
```diff
@@ -408,8 +410,37 @@ slot {
...
+    height: var(--auro-size-500, 40px);
+
+    ::slotted(auro-icon) {
+      width: var(--auro-size-300, 24px);
+      height: var(--auro-size-300, 24px);
+    }
+  }
+
+  // auro-button--hideText
+  &--hideText {
```
Comment (jason-capsule42) [inline src/style.scss:428]: Why not just include this CSS down below as part of the selector on line 436?

File: web-test-runner.config.mjs [hunk 9]
```diff
@@ -4,7 +4,7 @@ export default {
   coverageConfig: {
     threshold: {
       statements: 80,
-      branches: 70,
+      branches: 65,
```
Comment (jason-capsule42) [inline web-test-runner.config.mjs:7]: Rather than doing this let's just add the test for registering a custom name for a component and rendering that. This will bring the branches coverage to the required minimum.

File: src/auro-button.js [hunk 10]
```diff
@@ -28,12 +30,15 @@ import '@alaskaairux/auro-loader';
...
  * @prop {Boolean} ready - When false the component API should not be called.
  * @fires auroButton-ready - Notifies that the component has finished initializing.
+ * @fires auroButton-toggledText - Notifies that the visibility of the default slot content has changed.
  *
- * @slot - Provide text for the button.
+ * @slot - Default slot for the text of the button.
+ * @slot icon - Slot to provide auro-icon for the button.
  */
+
+/* eslint-disable max-statements, no-negated-condition */
```
Comment (Izook) [inline src/auro-button.js:39]: why did we disable these rules?

File: src/style.scss [hunk 11]
```diff
@@ -1,12 +1,14 @@
-// Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license
-// See LICENSE in the project root for license information.
+/* Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license */
+/* See LICENSE in the project root for license information. */
 
-// ---------------------------------------------------------------------
+/* --------------------------------------------------------------------- */
 
-// Import Auro Sass variables
+/* stylelint-disable at-rule-no-unknown */
```
Comment (Izook) [inline src/style.scss:6]: why did we disable this rule?

File: test/auro-button.test.js [hunk 12]
```diff
@@ -30,6 +32,12 @@ describe('auro-button', () => {
     expect(classList.includes('auro-button')).to.be.true;
   });
 
+  it('successfully registers custom component', async () => {
```
Comment (Izook) [inline test/auro-button.test.js:35]: Excellent job adding a test case for a previously added feature!

File: src/style.scss [hunk 13]
```diff
@@ -1,12 +1,14 @@
-// Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license
-// See LICENSE in the project root for license information.
+/* Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license */
```
Comment (blackfalcon) [inline src/style.scss:1]: Why are all these comments changed from Sass comments to CSS comments? There is no reason for this to be output into the CSS.

File: src/auro-button.js [hunk 14]
```diff
@@ -142,6 +161,12 @@ export class AuroButton extends LitElement {
 
   firstUpdated() {
     this.notifyReady();
+
+    this.addEventListener('auroButton-toggledText', () => {
```
Comment (Izook) [inline src/auro-button.js:165]: This may be me not understanding how Lit works but if I wanted to toggle `hideText` would I want to fire this event with the button as the target or would I update the `hideText` attribute? Are we supporting both avenues?

File: src/style.scss [hunk 15]
```diff
@@ -389,25 +392,52 @@ slot {
...
+:host([hideText][rounded]) {
+  ::slotted(*):not(auro-icon) {
+    display: none;
+  }
+
+  .auro-button {
+    min-width: unset;
+    padding: unset;
+
+    width: 48px;
```
Comment (blackfalcon) [inline src/style.scss:439]: This should be `var(--auro-size-600)`

File: src/style.scss [hunk 16]
```diff
@@ -389,25 +392,52 @@ slot {
...
     border-radius: 100px;
     min-width: unset;
   }
 
   &--iconOnlySlim {
     ::slotted(auro-icon) {
-      width: calc(var(--auro-size-300, 24px) - 4px);
-      height: calc(var(--auro-size-300, 24px) - 4px);
+      width: calc(var(--auro-size-300) - 4px);
+      height: calc(var(--auro-size-300) - 4px);
```
Comment (blackfalcon) [inline src/style.scss:409]: update -4px to `var(--auro-size-50)`

File: src/style.scss [hunk 17]
```diff
@@ -389,25 +392,52 @@ slot {
...
-      width: var(--auro-size-300, 24px);
-      height: var(--auro-size-300, 24px);
+      width: var(--auro-size-300);
+      height: var(--auro-size-300);
     }
 
-    padding-left: calc(var(--auro-size-100, 8px) - 2px);
-    padding-right: calc(var(--auro-size-100, 8px) - 2px);
+    padding-left: calc(var(--auro-size-100) - 2px);
+    padding-right: calc(var(--auro-size-100) - 2px);
```
Comment (blackfalcon) [inline src/style.scss:400]: Update -2px to `var(--auro-size-25)`

File: src/style.scss [hunk 18]
```diff
@@ -389,25 +392,52 @@ slot {
...
   &--iconOnlySlim {
     ::slotted(auro-icon) {
-      width: calc(var(--auro-size-300, 24px) - 4px);
-      height: calc(var(--auro-size-300, 24px) - 4px);
+      width: calc(var(--auro-size-300) - 4px);
+      height: calc(var(--auro-size-300) - 4px);
     }
 
     padding-left: var(--auro-size-50, 4px);
     padding-right: var(--auro-size-50, 4px);
```
Comment (blackfalcon) [inline src/style.scss:413]: This could be updated too.

File: package.json [hunk 19]
```diff
@@ -48,6 +48,7 @@
     "eslint-plugin-lit": "^1.5.1",
     "husky": "^8.0.0",
     "lodash": "^4.17.21",
+    "markdown-magic": "^2.6.1",
```
Comment (blackfalcon) [inline package.json:51]: Including this and the other updates to the build scripts should not be included in this commit. Adding markdown-magic has nothing to do with creating a rounded button. 

Please update the body of the commit message to include that included with the actual feature also adds the initial support for using Markdown Magic.

File: demo/index.min.js [hunk 20]
```diff
@@ -0,0 +1,5 @@
+// import { toggledTextExample } from "../apiExamples/toggledText";
+
+function initButtonIndexExamples(initCount) {
+}
```
Comment (blackfalcon) [inline demo/index.min.js:4]: Why is there an empty function?

File: src/auro-button.js [hunk 21]
```diff
@@ -12,11 +12,13 @@ import '@alaskaairux/auro-loader';
 /**
  * @attr {Boolean} autofocus - This Boolean attribute lets you specify that the button should have input focus when the page loads, unless overridden by the user
  * @attr {Boolean} disabled - If set to true button will become disabled and not allow for interactions
+ * @attr {Boolean} hideText - If set to true, hides default slot content when used in conjuction with the `rounded` attribute
```
Comment (blackfalcon) [inline src/auro-button.js:15]: This attribute is not showing up in the API doc?

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: demo/api.min.js [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+  const toggledTextExampleElemRight = document.getElementById('toggledTextExampleRight');
+
+  addEventListener(toggledTextExampleElemLeft);
+  addEventListener(toggledTextExampleElemRight);
+}
+
+function initButtonApiExamples(initCount) {
+  initCount = initCount || 0;
+
+  try {
```
Comment (Izook) [inline demo/api.min.js:18]: is there some event or something we can listen to instead of re-attempting the event listener addition multiple times?

File: demo/index.js [hunk 23]
```diff
@@ -0,0 +1,16 @@
+// import { toggledTextExample } from "../apiExamples/toggledText";
```
Comment (Izook) [inline demo/index.js:1]: should this be commented out?

File: demo/index.min.js [hunk 24]
```diff
@@ -0,0 +1,5 @@
+// import { toggledTextExample } from "../apiExamples/toggledText";
```
Comment (Izook) [inline demo/index.min.js:1]: same here?

File: src/style.scss (hunk 13 above)
Comment (jordanjones243) [reply src/style.scss:1]: Screen Shot 2023-06-08 at 6 24 22 PM

The comments were changed because this error was appearing in the console when running a build. The error was fixed by changing the comment type.

File: src/auro-button.js (hunk 14 above)
Comment (jordanjones243) [reply src/auro-button.js:161]: Writing an event for this is something that Jason suggested. I am not sure exactly if there is a benefit to having an event, but yes we are currently supporting both options.

File: src/style.scss (hunk 11 above)
[the remark this answers is not in the document]
Comment (jordanjones243) [reply src/style.scss:6]: This rule was disabled because there was an error in the console saying that the file was using `@` rules that were unknown to the linter. These `@` rules being used are common selectors used across multiple components, so I decided to disable the rule as we need these selectors.

File: src/auro-button.js (hunk 10 above)
Comment (jordanjones243) [reply src/auro-button.js:39]: The constructor has more statements than allowed by eslint and we use a negated condition to toggle the `hideText` attribute.

File: demo/api.min.js (hunk 22 above)
Comment (jordanjones243) [reply demo/api.min.js:18]: That is a great question and something I can bring to Jason's attention.

File: src/auro-button.js (hunk 10 above)
Comment (Izook) [reply src/auro-button.js:39]: Hmm the constructor rule breaking is hard to avoid. I'm not sure if auro has a pattern for ignoring these kinds of rules but it might make sense to [only ignore top level functions for now](https://eslint.org/docs/latest/rules/max-statements#ignoretoplevelfunctions)?

The negated conditional however appears to be in line 206 and easy to avoid if we use `this.hideText` as the condition instead of `!this.hideText`.
image

File: src/style.scss (hunk 11 above)
[the remark this answers is not in the document]
Comment (Izook) [reply src/style.scss]: Instead of disabling the rule we might be able to keep some linting functionality by [adding the unknown `@` rules to the config so they are no longer unknown](https://stylelint.io/user-guide/rules/at-rule-no-unknown/#optional-secondary-options)

File: src/auro-button.js (hunk 14 above)
Comment (Izook) [reply src/auro-button.js:161]: @jason-capsule42 can you provide some of your wisdom on this pattern?

File: src/style.scss (hunk 13 above)
Comment (Izook) [reply src/style.scss:1]: I'm noticing other repos provide certain extensions and plugins to support `.scss` styling ([auro-menu](https://github.com/AlaskaAirlines/auro-menu/blob/main/.stylelintrc#L2), [auro-datepicker](https://github.com/AlaskaAirlines/auro-datepicker/blob/main/.stylelintrc#L2), [auro-hyperlink](https://github.com/AlaskaAirlines/auro-hyperlink/blob/main/.stylelintrc#L3)), we may want to include these rules here for proper commenting. 

(this may also solve the unknown `@` rule issue??)

File: demo/api.min.js (hunk 22 above)
Comment (Izook) [reply demo/api.min.js:18]: If we just want to wait for the content to load some googling is showing we should be able to use the [window on load event](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event)? Might be worth trying out?

Comment (jordanjones243) [reply demo/api.min.js:18]: I think the main issue is that we need to wait for all of the html to load before we run the javascript. I don't think everything loads right when the window does.

@jason-capsule42 What do you think of Isaac's comment/idea?

Comment (jason-capsule42) [reply demo/api.min.js:18]: @Izook this is great idea to explore further. However, I'd like to get the button release out using the pattern we currently have and then let's dive deeper into how this could work for all auro-components as a separate ticket. Could you open a new ticket in the generator for this work?

File: src/auro-button.js (hunk 14 above)
Comment (jason-capsule42) [reply src/auro-button.js:161]: @jordanjones243 let's pair program on this. This is a different thing then what I was suggesting.

**Review by jason-capsule42 — DISMISSED** [review]

File: README.md [hunk 25]
```diff
@@ -1,149 +1,198 @@
...
-
-`<auro-button>` is responsive by default. The button will assume 100% of the width of its container for views less than [auro_breakpoint--sm](https://alaskaairlines.github.io/WebCoreStyleSheets/#responsive-mixin-auro_breakpoint--sm).
-
-Beyond that breakpoint `<auro-button>` will assume the width of the content or a min-width of `8.75rem`, which ever is greater.
-
-If the desired appearance of the `<auro-button>` is to be placed in the reverse direction of natural content, then the attributes of `responsive` and `reverse` are needed on the `<auro-button>` element.
-
-## Multi button support
-
-When the UI requires the use of multiple buttons within the same space, with the use of the [Auro Web Core Style Sheets](https://alaskaairlines.github.io/WebCoreStyleSheets/#utility-auro-css-#{$scope}.auro_containedButtons), and the `auro_containedButtons` .
```
Comment (blackfalcon) [inline README.md:72]: This should be left in the README

File: README.md [hunk 26]
```diff
@@ -1,149 +1,198 @@
...
-
-**NOTE:** Be sure to replace `@latest` in the URL with the version of the asset you want. @latest is NOT aware of any MAJOR releases, use at your own risk.
-
-## Responsive support
-
-`<auro-button>` is responsive by default. The button will assume 100% of the width of its container for views less than [auro_breakpoint--sm](https://alaskaairlines.github.io/WebCoreStyleSheets/#responsive-mixin-auro_breakpoint--sm).
-
-Beyond that breakpoint `<auro-button>` will assume the width of the content or a min-width of `8.75rem`, which ever is greater.
-
-If the desired appearance of the `<auro-button>` is to be placed in the reverse direction of natural content, then the attributes of `responsive` and `reverse` are needed on the `<auro-button>` element.
```
Comment (blackfalcon) [inline README.md:68]: This should remain in the README

**Comment by blackfalcon** [comment]
@jordanjones243 the .npmignore was missed with the past update, please update to the following. 

```
# standard ignore
*config*
*conf*
LICENSE
NOTICE

.github/
.gitignore
.eslintrc
.stylelintrc

coverage/
docs/
scripts/
test/
apiExamples/
docTemplates/
demo/**/*.css
demo/**/*.scss
demo/**/*.html

dist/es5.js
dist/auro-*.js
!dist/auro-*__bundled.js
dist/style-css.js

index.html

# ignore src
src/**/*.scss
src/**/*.css
```

**Comment by blackfalcon** [comment]
@jordanjones243 also update the .gitignore to the following 

```
.DS_Store
node_modules/
docTemplates/
.cache/

## Demo support files
demo/*.css
demo/*.css.map

## CSS built from Sass process
src/*.css
src/*.css.map

## *-css built from sass-render process
*-css.js

## dist files
dist/
build/

## Code coverage reports
coverage/
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: README.md (hunk 25 above)
Comment (jordanjones243) [reply README.md:72]: This is stuff that was automatically removed during `build`. Do you want me to manually reinsert this as well as the content from the comment below?

Comment (blackfalcon) [reply README.md:72]: Yes please.

File: .stylelintrc [hunk 27]
```diff
@@ -4,6 +4,7 @@
     "no-duplicate-selectors": null,
     "font-family-no-duplicate-names": null,
     "no-descending-specificity": null,
-    "selector-type-no-unknown": null
+    "selector-type-no-unknown": null,
+    "at-rule-no-unknown": null
```
Comment (jason-capsule42) [inline .stylelintrc:8]: Why was this added here rather than to the file or document that failed the lint rule?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
We need an additional example that shows how the text can be hidden/shown via custom interaction.

File: src/style.scss (hunk 13 above)
Comment (blackfalcon) [reply src/style.scss:1]: The problem is that for some reason not all the dependencies are listed? In the Generator package template there are the following: 

```
    "stylelint": "",
    "stylelint-config-idiomatic-order": "",
    "stylelint-config-standard": "",
    "stylelint-config-standard-scss": "",
    "stylelint-order": "",
    "stylelint-scss": "",
```

In this project, all there is

```
    "stylelint": "^15.7.0",
    "stylelint-config-standard": "^33.0.0",
```

Update these dependencies and then also update the .stylelintrc based on the configs from the Generator. This will also require updates to the code itself to meet the new standards. 

Although, I am ok with NOT updating to use `stylelint-config-idiomatic-orde`.

File: package.json [hunk 28]
```diff
@@ -79,13 +80,15 @@
...
+    "build": "npm-run-all build:sass sass:render scssLint dist:js build:docs bundler postinstall build:api build:demoScripts types",
     "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
     "build:api": "wca analyze 'src/auro-button.js' --outFiles docs/api.md",
+    "build:docs": "node scripts/generateDocs.js",
+    "build:demoScripts": "node scripts/prepExampleFiles.js",
     "build:watch": "nodemon -e scss,js --watch src --exec npm run build:dev:assets",
     "bundler": "rollup -c",
     "bundle:test": "rollup -c -w",
-    "scssLint": "stylelint \"./src/*.css\"",
+    "scssLint": "stylelint \"./src/**/*.scss\"",
```
Comment (blackfalcon) [inline package.json:91]: See notes about updating to SCSS linting dependencies

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: src/style.scss (hunk 13 above)
Comment (jordanjones243) [reply src/style.scss]: New commit was added to fix this issue

File: apiExamples/icon.html [hunk 29]
```diff
@@ -0,0 +1,12 @@
+<auro-button aria-label="wifi">
+  Activate WiFi
+  <auro-icon customColor category="in-flight" name="wifi" slot="icon"></auro-icon>
```
Comment (blackfalcon) [inline apiExamples/icon.html:3]:
```
  <auro-icon customColor category="in-flight" name="wifi" slot="icon"></auro-icon>
```

What is `slot="icon"` for?

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: apiExamples/icon.html (hunk 29 above)
Comment (blackfalcon) [reply apiExamples/icon.html:3]: Answered my own question. That is some old API stuff there. An icon as content works both in the main slot and the icon-specific slot. Ok then.

**Review by blackfalcon — APPROVED** [review]
:shipit:

**Review by blackfalcon — CHANGES_REQUESTED** [review]
The feature where you hover over a button and it shows content is not accessible. 

The concept of `hover` is not accessible either. There needs to be audible feedback as to the content of the button on a `tab` interaction. 

Screen Shot 2023-06-30 at 5 18 32 PM

**Review by blackfalcon — DISMISSED** [review]

File: .stylelintrc [hunk 30]
```diff
@@ -1,9 +1,109 @@
...
+  "extends": ["stylelint-config-recommended", "stylelint-config-standard-scss"],
+  "plugins": [
+    "stylelint-scss"
+  ],
   "rules": {
-    "no-duplicate-selectors": null,
-    "font-family-no-duplicate-names": null,
-    "no-descending-specificity": null,
-    "selector-type-no-unknown": null
+    "alpha-value-notation": "number",
```
Comment (Izook) [inline .stylelintrc:7]: Are all these rules specific to `auro-button` if not (just an idea) this might be a great addition to add the `auro-library` to standardize our styleling styles

File: src/style.scss [hunk 31]
```diff
@@ -1,14 +1,30 @@
...
-
-/* stylelint-disable at-rule-no-unknown */
-
-/* Import Auro Sass variables */
+// Copyright (c) Alaska Air. All right reserved. Licensed under the Apache-2.0 license
+// See LICENSE in the project root for license information.
+
+// ---------------------------------------------------------------------
+
+/* stylelint-disable
```
Comment (Izook) [inline src/style.scss:6]: Do we need to ignore all these rules and if so should we add them to the `.stylelintrc`?

File: apiExamples/roundedRightAlign.html [hunk 32]
```diff
@@ -0,0 +1,4 @@
+<auro-button aria-label="in-flight" rounded iconOnly id="rightAlignElem" style="float: right;">
+  Back to Top
```
Comment (jason-capsule42) [inline apiExamples/roundedRightAlign.html:2]: Use the same text for the button as the right align style.

File: apiExamples/roundedRightAlign.js [hunk 33]
```diff
@@ -0,0 +1,11 @@
+export function roundedRightAlignExample() {
+  const rightAlignElem = document.querySelector('#rightAlignElem');
+
+  rightAlignElem.addEventListener('mouseover', () => {
+    rightAlignElem.iconOnly = false;
+  });
+
+  rightAlignElem.addEventListener('mouseout', () => {
+    rightAlignElem.iconOnly = true;
+  });
```
Comment (jason-capsule42) [inline apiExamples/roundedRightAlign.js:10]: Include the focus events from the left align example as well.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/style.scss (hunk 31 above)
Comment (jordanjones243) [reply src/style.scss:6]: We do need to ignore all of these rules as of now until this repo gets a refactor. I was told that we should not edit the `.stylelintrc` as it is auto-generated from the generator.

File: .stylelintrc (hunk 30 above)
Comment (jordanjones243) [reply .stylelintrc:7]: Please see my comment below

File: apiExamples/roundedRightAlign.html (hunk 32 above)
Comment (jordanjones243) [reply apiExamples/roundedRightAlign.html]: Changes were made under this commit: [docs(rounded): add examples for rounded button](https://github.com/AlaskaAirlines/auro-button/pull/214/commits/3ba91aa69416d83385db790b3cde6264023b3f42) https://github.com/AlaskaAirlines/auro-button/issues/212

File: apiExamples/roundedRightAlign.js (hunk 33 above)
Comment (jordanjones243) [reply apiExamples/roundedRightAlign.js:11]: Changes were made under this commit: [docs(rounded): add examples for rounded button](https://github.com/AlaskaAirlines/auro-button/pull/214/commits/3ba91aa69416d83385db790b3cde6264023b3f42) https://github.com/AlaskaAirlines/auro-button/issues/212

File: src/style.scss (hunk 31 above)
Comment (Izook) [reply src/style.scss:6]: I see, that makes sense! I think in the future it would be nice to consolidate a lot of these rules to the `auro-library` and then we can use the local versions to override certain rules per-repo but we can discuss this another time!

**Review by blackfalcon — APPROVED** [review]

**Review by jason-capsule42 — APPROVED** [review]
